lib/rubocop/cop/rspec/subject_declaration.rb

```diff
@@ -25,7 +25,7 @@ class SubjectDeclaration < Base
 
         # @!method offensive_subject_declaration?(node)
         def_node_matcher :offensive_subject_declaration?, <<~PATTERN
-          (send nil? ${#Subjects.all #Helpers.all} {(sym :subject) (str "subject")} ...)
+          (send nil? ${#Subjects.all #Helpers.all} {(sym #Subjects.all) (str {"subject" | "subject!"})} ...)
```

Comment:
I don't think subject!(:subject!)` is technically offensive, as it defines the `subject!` instance method. Otherwise you can't use `subject!` in the spec.
Though this is open for interpretation. @pirj WDYT?

Reply:
It is unclear to me why would anyone use `subject!(:subject!)`, but if they would, I would consider it just as offensive as `subject(:subject)` or a nameless subject. Those two would be defined in different scopes, example group and example ones.

Reply:
I'm not sure what to do with this.
On one hand, it extends the cop, so if someone says  in the language config `instance` is a subject, it would look for `let(:instance)`. But it won't trigger on `let("instance")`. Perhaps it should be
```suggestion
          (send nil? ${#Subjects.all #Helpers.all} {(sym #Subjects.all) (str #Subjects.all.map(&:to_s))} ...)
```